unstructured/chunking/title.py

```diff
@@ -133,7 +132,25 @@ def chunk_by_title(
                     _value.extend(item for item in value if item not in _value)
                     setattr(metadata, attr, _value)
 
-        chunked_elements.append(CompositeElement(text=text, metadata=metadata))
+        # Check if text exceeds max_characters
+        if len(text) > max_characters:
+            # Chunk the text from the end to the beginning
+            while len(text) > 0:
+                if len(text) <= max_characters:
+                    # If the remaining text is shorter than max_characters
+                    # create a chunk from the beginning
+                    chunk_text = text
+                    text = ""
+                else:
+                    # Otherwise, create a chunk from the end
+                    chunk_text = text[-max_characters:]
+                    text = text[:-max_characters]
+
+                # Prepend the chunk to the beginning of the list
+                chunked_elements.insert(0, CompositeElement(text=chunk_text, metadata=metadata))
+        else:
+            # If it doesn't exceed, create a single CompositeElement
+            chunked_elements.append(CompositeElement(text=text, metadata=metadata))
```

Comment:
You'll have to verify this behavior matches what you're expecting but it passes tests. I think you can reduce this to a 2 liner 
```
chunks = [text[i:i + max_characters] for i in range(0, len(text), max_characters)]
chunked_elements.extend(CompositeElement(text=chunk, metadata=metadata) for chunk in chunks)
```

Reply:
I think this is almost what we want! This is iterating over text forwards, so if there's a leftover chunk smaller than max_characters it'll end up at the end. The behavior Crag and I talked about was it ending up at the start (so iterating over text backwards).

This is how I tweaked your improvement, but it's a little hairy to look at. Not sure it's worth the fewer lines of code:
`chunks = [text[max(0, i - max_characters + 1) : i + 1] for i in range(len(text) -1, -1, -max_characters)]`

Reply:
Great, feel free to leave as is then.